Kernel/Storage/SD/SDHostController.cpp

```diff
@@ -93,6 +98,37 @@ ErrorOr<void> SDHostController::initialize()
     return {};
 }
 
+void SDHostController::try_enable_dma()
```

Comment:
IMO it's a bit weird having to separately attempt to enable DMA. From the perspective of the user of SDHostController's interface, you would always call this since DMA is just better than PIO right?


I would just make this private and call it inside `initialize`. If the reason is the Raspberry Pi port doesn't have the DMA engine enabled yet, then I would just add a `#if !(ARCH(AARCH64))` before the call and slap a FIXME saying to remove it once it works

Reply:
being able to enable this is bus dependant,
So I think it should be the respnsibility of the bus specific driver class to check if it can allow dma
(For example for USB-SDHCs, which have different DMA restrictions)

Reply:
not sure how is this ‘tryingʼ. it just enables it? 😅

consider changing the name to `enable_dma`?

Reply:
Well it's trying in the sense that it has to check if it can enable it first
any third opinion on this to tie-break?

Reply:
nope! leave as is when in doubt! 😄

Reply:
I think the name is fine